code/modules/mining/gemstones.dm

```diff
@@ -0,0 +1,639 @@
+	icon_state = "lruby_1"
+	gemtype = GEM_RUBY
+	gemsize = GEM_LARGE
+	maxstack = 5
+	stacksize = 1
+
+/obj/item/precious/gemstone/topaz/large
+	name = "large topaz"
+	desc = "A single yellow gemstone of large size."
+	icon_state = "ltopaz_1"
+	gemtype = GEM_TOPAZ
+	gemsize = GEM_LARGE
+	maxstack = 5
+	stacksize = 1
+
+/obj/item/precious/gemstone/sapphire/large
+	name = "large sapphire"
+	desc = "A single blue gemstone of large size."
+	icon_state = "lsapphire_1"
+	gemtype = GEM_SAPPHIRE
+	gemsize = GEM_LARGE
+	maxstack = 5
+	stacksize = 1
+
+/obj/item/precious/gemstone/amethyst/large
+	name = "large amethyst"
+	desc = "A single purple gemstone of large size."
+	icon_state = "lamethyst_1"
+	gemtype = GEM_AMETHYST
+	gemsize = GEM_LARGE
+	maxstack = 5
+	stacksize = 1
+
+/obj/item/precious/gemstone/emerald/large
+	name = "large emerald"
+	desc = "A single green gemstone of large size."
+	icon_state = "lemerald_1"
+	gemtype = GEM_EMERALD
+	gemsize = GEM_LARGE
+	maxstack = 5
+	stacksize = 1
+
+/obj/item/precious/gemstone/diamond/large
+	name = "large diamond"
+	desc = "A single colorless gemstone of large size."
+	icon_state = "ldiamond_1"
+	gemtype = GEM_DIAMOND
+	gemsize = GEM_LARGE
+	maxstack = 5
+	stacksize = 1
+
+// Enormous Gemstones
+
+/obj/item/precious/gemstone/ruby/huge
+	name = "enormous ruby"
+	desc = "A single red gemstone of enormous size."
+	icon_state = "eruby"
+	gemtype = GEM_RUBY
+	gemsize = GEM_ENORMOUS
+	maxstack = 5
+	stacksize = 1
+
+/obj/item/precious/gemstone/topaz/huge
+	name = "enormous topaz"
+	desc = "A single yellow gemstone of enormous size."
+	icon_state = "etopaz"
+	gemtype = GEM_TOPAZ
+	gemsize = GEM_ENORMOUS
+	maxstack = 5
+	stacksize = 1
+
+/obj/item/precious/gemstone/sapphire/huge
+	name = "enormous sapphire"
+	desc = "A single blue gemstone of enormous size."
+	icon_state = "esapphire"
+	gemtype = GEM_SAPPHIRE
+	gemsize = GEM_ENORMOUS
+	maxstack = 5
+	stacksize = 1
+
+/obj/item/precious/gemstone/amethyst/huge
+	name = "enormous amethyst"
+	desc = "A single purple gemstone of enormous size."
+	icon_state = "eamethyst"
+	gemtype = GEM_AMETHYST
+	gemsize = GEM_ENORMOUS
+	maxstack = 5
+	stacksize = 1
+
+/obj/item/precious/gemstone/emerald/huge
+	name = "enormous emerald"
+	desc = "A single green gemstone of enormous size."
+	icon_state = "eemerald"
+	gemtype = GEM_EMERALD
+	gemsize = GEM_ENORMOUS
+	maxstack = 5
+	stacksize = 1
+
+/obj/item/precious/gemstone/diamond/huge
+	name = "enormous diamond"
+	desc = "A single colorless gemstone of enormous size."
+	icon_state = "ediamond"
+	gemtype = GEM_DIAMOND
+	gemsize = GEM_ENORMOUS
+	maxstack = 5
+	stacksize = 1
+
+// Temporary treasure hunt space - to be deleted after someone finds the first enormous gemstone
+
+/obj/item/precious/gemstone/ruby/huge/pickup(mob/living/user)
+	if(src.pickup == 0)
+		to_chat(world, "<span style='color: green;'><i>[usr.name] has found the first ever enormous gemstone - a huge ruby! You should inform Kaedwuff OOC for a surprise.</i></span>")
```

Comment:
Yeah, no, I'm not a fan of this.

Reply:
How about I just make it message the first person that picks it up.

Reply:
That won't work either. I think this should be completely removed really. If anything just remark to the person that it's super rare.

Reply:
There, a compromise.  Only the admins will get the notice.